### PR TITLE
Converted preprocessor meta examples

### PR DESCRIPTION
**Pull request by avramidis**
The are two more undocumented examples that I didn't convert because I need the changes from #4423 
If the changes here are fine I can later add the generated data files.

### CODE REVIEW

File: src/shogun/kernel/Chi2Kernel.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ using namespace shogun;
 void
 CChi2Kernel::init()
 {
-	SG_ADD(&width, "width", "Kernel width.", ParameterProperties::HYPER);
+	SG_ADD(&width, "kernel_width", "Kernel width.", ParameterProperties::HYPER);
```
Comment (karlnapf) [inline src/shogun/kernel/Chi2Kernel.cpp]: why this?

File: src/shogun/preprocessor/FisherLDA.h [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
 #include <shogun/features/Features.h>
 #include <shogun/labels/Labels.h>
+#include <shogun/labels/MulticlassLabels.h>
```
Comment (karlnapf) [inline src/shogun/preprocessor/FisherLDA.h]: not sure why this is needed?

File: src/shogun/kernel/Chi2Kernel.cpp (hunk 1 above)
Comment (avramidis) [reply src/shogun/kernel/Chi2Kernel.cpp]: Was getting and error that was saying it was double defined

File: src/shogun/preprocessor/FisherLDA.h (hunk 2 above)
Comment (avramidis) [reply src/shogun/preprocessor/FisherLDA.h]: Could not compile differently. I am not sure why

**Comment by avramidis** [comment]
Great! I will add the data files too.

File: src/shogun/kernel/Chi2Kernel.cpp (hunk 1 above)
Comment (karlnapf) [reply src/shogun/kernel/Chi2Kernel.cpp]: ok

**Comment by karlnapf** [comment]
could you rebase this, as there is a merge conflict right now (data was changed in another commit)

File: examples/meta/src/preprocessor/fishlda.sg [hunk 3]
```diff
@@ -0,0 +1,18 @@
+File data = csv_file("../../data/fm_train_real.dat")
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/fishlda.sg]: could you name this "fisher_lda"?

**Comment by avramidis** [comment]
Ok, no problem.

File: examples/meta/src/preprocessor/kernelpca.sg [hunk 4]
```diff
@@ -0,0 +1,21 @@
+File data = csv_file("../../data/fm_train_real.dat")
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/kernelpca.sg]: rename to "kernel_pca"

File: examples/meta/src/preprocessor/logplusone.sg [hunk 5]
```diff
@@ -0,0 +1,30 @@
+File data_train = csv_file("../../data/fm_train_real.dat")
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/logplusone.sg]: rename to "log_plus_one" ... you get the point of how we want the filenames to look like :)

File: examples/meta/src/preprocessor/randomfouriergausspreproc.sg [hunk 6]
```diff
@@ -0,0 +1,32 @@
+File data_train = csv_file("../../data/fm_train_real.dat")
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/randomfouriergausspreproc.sg]: random_fourier_features.sg

File: src/shogun/kernel/Chi2Kernel.cpp (hunk 1 above)
Comment (karlnapf) [reply src/shogun/kernel/Chi2Kernel.cpp]: Actually, I have a hard time believing that there was no single test that ran the kernel constructor before ...

EDIT. I just checked, none of the base classes register a parameter called `width`, so why would this error appear?

File: src/shogun/preprocessor/FisherLDA.h (hunk 2 above)
Comment (karlnapf) [reply src/shogun/preprocessor/FisherLDA.h]: I have doubts here as well.... the CI never complained about this but we actually have some test for fisher LDA somewhere so it was compiled before

EDIT: Just checked, the class is used but not included. Could you forward declare it instead?

File: src/shogun/kernel/Chi2Kernel.cpp (hunk 1 above)
Comment (avramidis) [reply src/shogun/kernel/Chi2Kernel.cpp]: I am not sure. I will investigate.

Comment (avramidis) [reply src/shogun/kernel/Chi2Kernel.cpp]: Changed it back and now it works with "width". Honestly, I don't know why I had an error before...

**Comment by avramidis** [comment]
Ah yes, I know why it fail. Will fix it tomorrow. I forgot to debase why data PR...

File: src/shogun/kernel/Chi2Kernel.cpp (hunk 1 above)
Comment (karlnapf) [reply src/shogun/kernel/Chi2Kernel.cpp]: lol well I guess nevermind then :)

**Comment by karlnapf** [comment]
rebase and merge?

**Comment by avramidis** [comment]
Rebased and tests passed. We could merge it if it's ok with you.

File: examples/meta/src/preprocessor/fish_lda.sg [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+#![create_pca_preprocessor]
+Transformer preproc = transformer("FisherLDA")
+preproc.put("final_dimensions", 1)
+preproc.put("method", enum EFLDAMethod.CANVAR_FLDA)
+preproc.fit(sg_features, sg_labels)
+#![create_pca_preprocessor]
+
+#![transform_features]
+Features featstrans = preproc.transform(sg_features)
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/fish_lda.sg]: Just he one more look.
There is the thing that only numerical values/vectors etc are serialised for the testing, not the objects themselves. So in this example here, the transformed features (matrix) are not part of the test. But I think they should

Comment (karlnapf) [reply examples/meta/src/preprocessor/fish_lda.sg]: So if you extract the feature matrix, it will be serialised

Comment (avramidis) [reply examples/meta/src/preprocessor/fish_lda.sg]: ok

File: examples/meta/src/preprocessor/fish_lda.sg [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+preproc.put("method", enum EFLDAMethod.CANVAR_FLDA)
+preproc.fit(feats_train, feats_labels)
+#![create_preprocessor]
+
+#![transform_features]
+Features feats_traing_trans = preproc.transform(feats_train)
+#![transform_features]
+
+#![get_output]
+RealMatrix feats_train_trans_matrix = feats_traing_trans.get_real_matrix("feature_matrix")
```
Comment (karlnapf) [inline examples/meta/src/preprocessor/fish_lda.sg]: ++
No need to have a marker (`get_output`) as this is not referenced in a cookbook, but no need to change now

**Comment by avramidis** [comment]
There is a problem with the pca results. The cause is the different sign. I could manually fix that so it can pass the CI tests because I cannot do my machine to return the same result as the CI ones. What do you think?

**Comment by karlnapf** [comment]
CI now fails, not sure what is happening there, should be passing if you add the files. Maybe forgot to update the data submodule?

**Comment by karlnapf** [comment]
ah we wrote at the same time ...

**Comment by karlnapf** [comment]
give the manual fix a try. But I bet this has to do with the eigensolver backend. We had sign issues with those before ...

**Comment by avramidis** [comment]
Yes, that is the problem. I am trying different libraries all day but no luck. I will give a try the manual fix.

**Comment by karlnapf** [comment]
ah this is annoying, it is a known problem with the integration tests.... I am not sure we can even fix this. In the unit tests we make the tests invariant for sign...but meta examples we cannot do this.
One option is to make the CI pass and then use those as reference (while they fail for other archs)... maybe give it a try

**Comment by karlnapf** [comment]
If the manual fix doesnt work, don't loose sleep. Then we need to make the eigensolvers signs platform independent first

**Comment by avramidis** [comment]
OK, test passed now.

**Comment by karlnapf** [comment]
Ok let’s see how this goes on the buildbot....

**Pull request merged**